src/main/java/org/wso2/carbon/securevault/aws/secret/handler/AWSSecretCallbackHandler.java

```diff
@@ -0,0 +1,121 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.securevault.aws.secret.handler;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.securevault.aws.exception.AWSSecretCallbackHandlerException;
+import org.wso2.carbon.securevault.aws.secret.repository.AWSSecretRepository;
+import org.wso2.securevault.secret.AbstractSecretCallbackHandler;
+import org.wso2.securevault.secret.SingleSecretCallback;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.CONFIG_FILE_PATH;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.IDENTITY_KEY_PASSWORD_ALIAS;
+import static org.wso2.carbon.securevault.aws.common.AWSVaultConstants.IDENTITY_STORE_PASSWORD_ALIAS;
+
+/**
+ * Secret Callback handler class if keystore and primary key passwords are stored in the AWS Vault.
+ */
+public class AWSSecretCallbackHandler extends AbstractSecretCallbackHandler {
+
+    private static final Log log = LogFactory.getLog(AWSSecretCallbackHandler.class);
+    private static String keyStorePassword;
+    private static String privateKeyPassword;
+
+    /**
+     * Handles single secret callback.
+     *
+     * @param singleSecretCallback A single secret callback.
+     */
+    @Override
+    protected void handleSingleSecretCallback(SingleSecretCallback singleSecretCallback) {
+
+        if (StringUtils.isEmpty(keyStorePassword) && StringUtils.isEmpty(privateKeyPassword)) {
+            boolean sameKeyAndKeyStorePass = true;
+            String keyPassword = System.getProperty("key.password");
+            if (keyPassword != null && keyPassword.trim().equals("true")) {
+                sameKeyAndKeyStorePass = false;
+            }
+            readPassword(sameKeyAndKeyStorePass);
+        }
+
+        if (singleSecretCallback.getId().equals("identity.key.password")) {
+            singleSecretCallback.setSecret(privateKeyPassword);
+        } else {
+            singleSecretCallback.setSecret(keyStorePassword);
+        }
+
+    }
+
+    /**
+     * Reads keystore and primary key passwords from AWS Vault.
+     *
+     * @param sameKeyAndKeyStorePass Flag to indicate whether the keystore and primary key passwords are the same.
+     */
+    @SuppressFBWarnings("PATH_TRAVERSAL_IN")
+    private void readPassword(boolean sameKeyAndKeyStorePass) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Reading configuration properties from file.");
+        }
+
+        Properties properties = new Properties();
+
+        //Reading configurations from file.
+        try (InputStream inputStream = new FileInputStream(CONFIG_FILE_PATH)) {
+            properties.load(inputStream);
+
```

Comment:
Remove unnecessary newline. Fix in other places too.

Reply:
Removed. Fixed in other places too.